src/components/ImmutableDevices/ImmutableDevices.cy.js

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { Route, Routes } from 'react-router-dom';
+import ImmutableDevices from './ImmutableDevices';
+import {
+  featureFlagsInterceptors,
+  hostsInterceptors,
+} from '../../../cypress/support/interceptors';
+
+const defaultProps = {
+  mergeAppColumns: (columns) => columns,
+};
+
+// eslint-disable-next-line react/prop-types
+const MockRouter = ({ path = '/insights/inventory', ...props }) => (
+  <Routes>
+    <Route path={path} element={<ImmutableDevices {...props} />} />
+    <Route path={'*'} element={<div id="mock-detail-page" />} />
+  </Routes>
+);
+
+const getEntities =
+  (dataInject) => async (items, config, showTags, defaultGetEntities) => {
+    const result = await defaultGetEntities(items, config, showTags);
+    result.results.map(dataInject);
+    return result;
+  };
+
+const mountWithProps = (props) => {
+  return cy.mountWithContext(
+    MockRouter,
+    {
+      routerProps: { initialEntries: ['/insights/inventory'] },
+    },
+    props
+  );
+};
+
+before(() => {
+  cy.mockWindowChrome();
+});
+
+describe('ImmutableDevices', () => {
+  beforeEach(() => {
+    cy.intercept('*', { statusCode: 200, body: { results: [] } });
+    hostsInterceptors.successful();
+    featureFlagsInterceptors.edgeParitySuccessful();
+  });
+
+  it('renders without issues', () => {
+    mountWithProps(defaultProps);
+
+    cy.get('table[aria-label="Host inventory"]').should(
+      'have.attr',
+      'data-ouia-safe',
+      'true'
+    );
```

Comment:
I am not sure asserting data-ouia-safe is enough to understand whether the table renders without problems. The thing is, the parameter is never updated in fact: https://www.patternfly.org/2022.05/components/table/react-legacy#table - it has default value set to true for ouiaSafe. And we also don't provide this parameter explicitly to PfTable (see EntityTable.js) which makes data-ouia-safe basically useless at the moment.

Reply:
Aha, will choose another selector.